extensions/Protobuf/ResponseReader.fs

```diff
@@ -0,0 +1,19 @@
+namespace Oryx.Protobuf
+
+open System.IO
+open System.Net.Http
+open System.Threading.Tasks
+open FSharp.Control.Tasks.V2.ContextInsensitive
+open Oryx
+
+module ResponseReaders =
+    let protobuf<'b, 'r, 'err> (parser : Stream -> 'b) (next: NextFunc<'b, 'r, 'err>) (context : Context<HttpResponseMessage>) : Task<Result<Context<'r>,HandlerError<'err>>> =
+        task {
+            let response = context.Response
+            let! stream = response.Content.ReadAsStreamAsync ()
+            try
+                let b = parser stream
+                return! next { Request = context.Request; Response = b }
+            with
+            | ex -> return Error (Panic ex)
+        }
```

Comment:
I'm not sure I quite understand what makes this function specific to protocol buffers? Seems pretty general to me.

Reply:
Yes, you are right. It takes a "generic" parser that matches the signature of what is generated by the protobuf compiler (`Stream -> 'b`). We could have re-used this for the other use-cases if the signature of the parser had been `Stream -> Task<'b>` instead.

Reply:
We could perhaps have had two generic functions `deserialize` and `deserializeAsync` or `parse` /  `parseAsync`. The JSON extension could partially apply `parseAsync` to get their `json` handler.

Reply:
Fixed!